src/server/services/Analytics.js

```diff
@@ -0,0 +1,101 @@
+require('dotenv').config();
+
+// Mongo setup
+const mongourl = process.env.DB_URL;
+const dbName = process.env.DB_DATABASE;
+const client = require('../models/db');
+
+let analytics = {};
+
+/**
+ * Returns Library analytics for a given class.
+ *
+ * @returns {object}   Analytics
+ */
+async function getAnalytics() {
+  await client.connect(mongourl);
+  const dbAnalytics = client.db(dbName);
+  const cursor = await dbAnalytics.collection('analytics').find();
+  const result = await cursor.toArray();
+
+  result.map(
+    x =>
+      (x.reserve_clicks =
+        x.reserve_clicks === undefined ? 0 : x.reserve_clicks.length)
+  );
+  result.map(
+    x =>
+      (x.research_clicks =
+        x.research_clicks === undefined ? 0 : x.research_clicks.length)
+  );
+  result.map(
+    x =>
+      (x.lib_tour_clicks =
+        x.lib_tour_clicks === undefined ? 0 : x.lib_tour_clicks.length)
+  );
+  result.map(
+    x =>
+      (x.research_tuts_clicks =
+        x.research_tuts_clicks === undefined
+          ? 0
+          : x.research_tuts_clicks.length)
+  );
+
+  result.map(
+    x =>
+      (x.total_reserve_clicks =
+        x.total_reserve_clicks === undefined ? 0 : x.total_reserve_clicks)
+  );
+  result.map(
+    x =>
+      (x.total_research_clicks =
```

Comment:
Not super happy with this area of code, but not sure how else to do it.

Reply:
is it possible to do result.map( x => { .... }) and then just put everything into that body?

Reply:
I tried that, it didn't work for some reason.

Reply:
@finnzink Can you put in a gist or codepen of what you tried? It should work. There is lots of code duplication and better to accomplish this in one pass than multiple loops

Reply:
@rlorenzo https://gist.github.com/finnzink/6c8e2639be25eef3763a8b0bf2946b92

Reply:
@finnzink What does result look like? I have this to try to test:

https://codepen.io/rlorenzo/pen/wvGoozW

Reply:
@rlorenzo @kilometerskang I got it to work, thanks for the feedback. Look at most recent commit.